### PR TITLE
feat: resize all images to the same aspect ratio and format

### LINKED ISSUE #699: Verify all our library logos are 640x480
Mentioned in the Dev Meeting - we should make sure all our images are the same size to avoid a weird-looking landing page

### PR DESCRIPTION
**Pull request by RobPasMue**
Implemented script to automate this process in the future whenever a new thumbnail is added. Transparent background will not be accepted. One time process script - instructions on README.

Closes #699

### CODE REVIEW

**Comment by RobPasMue** [comment]
Closes #699

File: doc/source/_static/thumbnails/resize.py [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+    """
+    # Open the input image
+    img = Image.open(input_image_path).convert("RGBA")  # Ensure the image has an alpha channel
+
+    # Resize the image while maintaining aspect ratio
+    img.thumbnail(target_size, Image.LANCZOS)
+
+    # Create a white background of the target size
+    background = Image.new(
+        "RGBA", target_size, (255, 255, 255, 255)
```
Comment (Revathyvenugopal162) [inline doc/source/_static/thumbnails/resize.py:26]:
```suggestion
        "RGBA", target_size, (255, 255, 255, 0)
```
why not initially remove the alpha value making the entire background initially transparent.

Comment (RobPasMue) [reply doc/source/_static/thumbnails/resize.py:26]: We need to make it white - i.e. remove transparency. This is on purpose.

**Review by germa89 — APPROVED** [review]
LGTM.

Quick questions:
- Why transparency is not allowed? I think all the images should be provided without background, and then, using your script, I would create two images, one with background for light theme and another for dark theme. Ideally we would like the users to provide two images (dark and light).. but if not, we could use the transparency stuff (if present), if not, then white/greish?.
- It does not appear in this PR, but is the script being executed as part of the CICD? or is the user who opened the PR who has to run it locally in the PC? Then you should probably check that all the files have that resolution in the CICD. I think that is just more work, I would just resize all the images in the doc build job.

**Comment by germa89** [comment]
I'm realising that the `grid-card-item` does not support two images per se... you can only specify one. Sad. :(

**Comment by RobPasMue** [comment]
> * Why transparency is not allowed?

Many images do not have the capability to provide a transparent background. We should homogenize the way things look in the landing page. This was brought up yesterday in the meeting... I'd have appreciated it you had mentioned it live 😄 We agreed on this since nobody complained.

> * It does not appear in this PR, but is the script being executed as part of the CICD? or is the user who opened the PR who has to run it locally in the PC? Then you should probably check that all the files have that resolution in the CICD. I think that is just more work, I would just resize all the images in the doc build job.

We can do it on CI/CD. That works for me. I'll implement that.

**Comment by germa89** [comment]
> Many images do not have the capability to provide a transparent background. We should homogenize the way things look in the landing page. This was brought up yesterday in the meeting... I'd have appreciated it you had mentioned it live 😄 We agreed on this since nobody complained.


Upss.. I'm very sorry about that! Maybe I was distracted for a bit 😅. IMHO I think transparency is the best ... after the option of having two different images (dark/light theme) but that is not an option with the `grid`.
I also think that if we want to homogenize the view, we should start by asking all the libraries to submit a proper logo... and not just simulation images.... or ask to all the libraries to submit simulation images. Then, talking about whether having or not transparent/light/dark background could have better context and it will definitely look better.

I dont mind either.. simulation, logo... but I think that whatever we put there, it should be dark/light theme compatible.

If you want the libraries to be compliant... then, "no logo, no pyansys homage" rule 🤣

Pinging @MaxJPRey @jorgepiloto

File: doc/source/_static/thumbnails/README.md [hunk 2]
```diff
@@ -0,0 +1,18 @@
+# Thumbnails
+
+This directory contains thumbnails for the documentation. They are used in the
+documentation to provide a visual representation of the various PyAnsys libraries
+and tools.
+
+## Resizing
+
+All thumbnails should be 640x480 pixels. This is the size that the documentation
+expects and is resized to this size if it is not already. Resize takes place
```
Comment (SMoraisAnsys) [inline doc/source/_static/thumbnails/README.md:10]:
```suggestion
All thumbnails should be 640x480 pixels. This is the size required by the
documentation. If they are not already this size, thumbnails will be resized to
640x480 pixels. Resize takes place
```
Minor suggestion to ease reading

**Comment by RobPasMue** [comment]
> > Many images do not have the capability to provide a transparent background. We should homogenize the way things look in the landing page. This was brought up yesterday in the meeting... I'd have appreciated it you had mentioned it live 😄 We agreed on this since nobody complained.
> 
> Upss.. I'm very sorry about that! Maybe I was distracted for a bit 😅. IMHO I think transparency is the best ... after the option of having two different images (dark/light theme) but that is not an option with the `grid`. I also think that if we want to homogenize the view, we should start by asking all the libraries to submit a proper logo... and not just simulation images.... or ask to all the libraries to submit simulation images. Then, talking about whether having or not transparent/light/dark background could have better context and it will definitely look better.
> 
> I dont mind either.. simulation, logo... but I think that whatever we put there, it should be dark/light theme compatible.
> 
> If you want the libraries to be compliant... then, "no logo, no pyansys homage" rule 🤣
> 
> Pinging @MaxJPRey @jorgepiloto

Can we do all this in a follow up issue? For now I'd like to get the homogeneity in place. Currently it's a mess... I'll move it to run on workflow but after that we should merge it. We can discuss about the transparency issue in a follow up.

**Review by SMoraisAnsys — APPROVED** [review]
LGTM
Would it be a good idea to add a check job in CI to ensure that every changes to doc/source/_static/thumbnails/*.png follows the requirement ?
Edit: just saw https://github.com/ansys/pyansys/pull/701#issuecomment-2363379467, glad we are all thinking alike haha

File: doc/source/_static/thumbnails/README.md (hunk 2 above)
Comment (RobPasMue) [reply doc/source/_static/thumbnails/README.md:10]: Thanks @SMoraisAnsys!

**Comment by RobPasMue** [comment]
> LGTM Would it be a good idea to add a check job in CI to ensure that every changes to doc/source/_static/thumbnails/*.png follows the requirement ?

Let's see what happens after https://github.com/ansys/pyansys/pull/701#issuecomment-2363574415 gets debated and decided. For now I will be moving this logic to the workflow/docs run so that they are dynamically adapted.

**Comment by greschd** [comment]
> asking all the libraries to submit a proper logo... 

That's something for the branding team, no? In any case, then I'd guess they'd want to make them somewhat uniform.. and consequently, the landing page would be boring. I think the "simulation / creative" image approach work better here.

**Comment by greschd** [comment]
Gray background works (IMO) in both themes: :wink: 

image

image

**Comment by RobPasMue** [comment]
> Gray background works (IMO) in both themes: 😉

Agreed but then teams have to come up with new images with gray background - which we are not going to tackle in this PR 😄 I'd prefer we have a new issue to discuss all this.

**Comment by RobPasMue** [comment]
I just opened #704 - let's keep the discussion there if possible and leave this PR for the aspect ratio alignment and white background (temporary solution).

**Pull request merged**